### ISSUE TITLE
How long does it take for training 2 epochs on 8 x NVIDIA A100

### ISSUE THREAD

**Issue by boogiegames3** [databrickslabs/dolly#171]
Hi:

I was able to successfully fine-tune with the training script you provided, using the `EleutherAI/pythia-12b` as the base model, but I am trying to verify that I did the same that you did. 

I am wondering how long is training supposed to run for for 8 x A100.

Can you share the tensorboard board to understand how the training process happened for the model that you published on HuggingFace?

Thanks in advance

**Comment by srowen** [comment]
1 epoch on 8 A100s is probably several hours. Depends a bit on what data you use and settings you might change. I don't think the TB logs are lying around from that run, but, just run the training script and watch metrics and progress to 0.01 epochs or something to extraploate.

**Comment by boogiegames3** [comment]
Interesting, the reason I ask is because seems to only be taking 1 hour for me, which seems suspect :-). I am using the default deepspeed config provided, the default dolly 15k fine-tuning dataset with the default configuration:
```
!deepspeed 8 \
     --module training.trainer \
     --input-model EleutherAI/pythia-12b \
     --deepspeed {deepspeed_config} \
     --epochs 2 \
     --local-output-dir {local_output_dir} \
     --dbfs-output-dir {dbfs_output_dir} \
     --per-device-train-batch-size 6 \
     --per-device-eval-batch-size 6 \
     --logging-steps 10 \
     --save-steps 200 \
     --save-total-limit 20 \
     --eval-steps 50 \
     --warmup-steps 50 \
     --test-size 200 \
     --lr 5e-6
```
<img width="541" alt="Screenshot 2023-05-17 at 8 00 01 AM" src="https://github.com/databrickslabs/dolly/assets/95264684/8ef932f6-898b-43ec-9ead-6f51fc4a8f82">


<img width="536" alt="Screenshot 2023-05-17 at 8 02 47 AM" src="https://github.com/databrickslabs/dolly/assets/95264684/b14b976e-46e1-422f-83f2-9c7c1fe675dd">

**Comment by boogiegames3** [comment]
No, the time for 2 full epochs. It took 1 hour for 2 epochs on 8 x A100 80GB.

**Comment by srowen** [comment]
Just double checking - that shows about 600 steps, which is 3600 inputs, which isn't nearly a full epoch. Is what's above what you mean or you're definitely sure it was 2 epochs (more like 5000 steps at batch size of 6). If so, then it's likely I just misremember, as I only tried this early on once

**Comment by boogiegames3** [comment]
That's what I mean.

<img width="551" alt="Screenshot 2023-05-17 at 3 47 13 PM" src="https://github.com/databrickslabs/dolly/assets/95264684/293d5360-776d-4b9d-ba3b-0adc9c1cf363">

Am I reading that graph incorrectly or is it wrong?

**Comment by boogiegames3** [comment]
I see.

Related question, the eval's loss trends upwards with the arguments from the train_dolly.py script, and also even when I make the test-size bigger. That's part of the reason why I was asking if you had the Tensorboard for the training of the published model, it would be great to be able to compare for reproducibility sake.

<img width="1644" alt="Screenshot 2023-05-18 at 9 44 26 AM" src="https://github.com/databrickslabs/dolly/assets/95264684/48ad3b34-8d12-471e-b8b5-f5b357087a74">

**Comment by srowen** [comment]
That looks familiar. Train loss drops at the end of each epoch and is mostly flat. Eval loss may or may not go down at first, but one quickly runs into overfitting with models like this and eval loss increases after an epoch or two. I'm not sure we had a great explanaiton. It seems, generally, like overfitting / memorization. Train loss is better once it starts seeing what it saw before in training. And so eval is getting worse quickly because of that. The answers would be more data or a smaller model really. 12B is just overkill, I think, unless you have a lot of fine-tuning data.

**Comment by boogiegames3** [comment]
Thank you.
